lib/util/async_store_listener.dart

```diff
@@ -5,21 +5,26 @@ import '../l10n/l10n_from_string.dart';
 import 'async_store.dart';
 import 'observer_consumers.dart';
 
-class AsyncStoreListener extends StatelessWidget {
-  final AsyncStore asyncStore;
+class AsyncStoreListener<T> extends StatelessWidget {
+  final AsyncStore<T> asyncStore;
   final String? successMessage;
+  final String Function(
+    BuildContext context,
+    AsyncStateData<T> asyncStore,
```

Comment:
*asyncState

Reply:
Actually this makes more sense:

```suggestion
    T data,
```